lib/wibox/widget/textclock.lua

```diff
@@ -1,6 +1,42 @@
 ---------------------------------------------------------------------------
 --- Text clock widget.
 --
+-- The `wibox.widget.textclock` widget is part of the Awesome WM's wiboxes
+-- system (see @{03-declarative-layout.md}).
+--
+-- This widget displays a text clock formated by the
+-- [GLib Date Time format](https://developer.gnome.org/glib/stable/glib-GDateTime.html#g-date-time-format)
+-- and [GTimeZone](https://developer.gnome.org/glib/stable/glib-GTimeZone.html#g-time-zone-new).
+--
+-- The `wibox.widget.textclock` inherits from `wibox.widget.textbox`. It means
+-- that, once created, the user will receive a derivated instance of
+-- `wibox.widget.textbox` associated with a private `gears.timer` to manage
+-- timed updates of the displayed clock.
+--
+-- Use a `wibox.widget.textclock`
+-- ---
+--
+-- @DOC_wibox_widget_defaults_textclock_EXAMPLE@
+--
+-- Alternatively, you can declare the `textclock` widget using the
+-- declarative pattern (Both codes are strictly equivalent):
+--
+-- @DOC_wibox_widget_declarative-pattern_textclock_EXAMPLE@
+--
+-- The GLib DateTime format
+-- ---
+--
+-- The time displayed by the textclock widget can be formated by the GLib
+-- DateTime format.
+--
+-- Here is a short list with commonly used format specifiers (extracted from
+-- the Glib API references):
+--
+--@DOC_glib_timedate_format_COMMON@
```

Comment:
Need to add space after the double dash. Otherwise the link below (g-date-time-format) isn't rendered properly

Reply:
Nope, with the additional space, the generated documentation will have the double dash drawn before the table. If you check how other files from `docs/common/*.ldoc` are included, they do the same.

It's a tricky hack to integrate the file content without making it a "comment of a comment" (output `----` vs `-- --`). ldoc will interpret `----` as a regular comment opening while `-- --` will be a comment followed by a double dash (to be rendered).

You can check that after running `make ldoc` by inspecting the `build/lib/wibox/widget/textclock.lua` file and checking the generated code that ldoc will parse to generate the API references.

Reply:
Thanks for this detail explanation, I wasn't aware of `build/lib/wibox/widget/textclock.lua` file (this simplifies my code investigation quite a lot).  Running `make ldoc` on the following snippet:
```
----<table class='widget_list' border=1>
-- </table>
--
-- You can read more on the GLib DateTime format at
-- [https://developer.gnome.org/glib/stable/glib-GDateTime.html#g-date-time-format](https://developer.gnome.org/glib/stable/glib-GDateTime.html#g-date-time-format).
```

Causes the 'html' output `build/doc/widgets/wibox.widget.textclock.html` to show the `https://developer.gnome.org/glib/stable/glib-GDateTime.html#g-date-time-format` part as simple text rather than a link. Looking at the html of this file I see:
```
 You can read more on the GLib DateTime format at
 [https://developer.gnome.org/glib/stable/glib-GDateTime.html#g-date-time-format](https://developer.gnome.org/glib/stable/glib-GDateTime.html#g-date-time-format).
```
(so `ldoc` didn't render the markdown at all)

Adding space to the `<table>` line:
```
---- <table class='widget_list' border=1>
```

make the link being rendered correctly. I don't know how to solve this problem atm (I did miss the extra dashes added by my proposed solution), but I'd look into it.

Is the link, in this commit, rendered correctly on your setup ? (maybe I have a version issue, as I often have with Ubutnu)